### PR TITLE
Add serverSecurityAlertPolicies APIs for MySQL servers

### PR DESCRIPTION
**Pull request by ayeletshpigelman**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 1]
```diff
@@ -1916,6 +2036,74 @@
...
+          "type": "boolean"
+        },
+        "storageEndpoint": {
+          "description": "Specifies the blob storage endpoint (e.g. https://MyAccount.blob.core.windows.net). This blob storage will hold all Threat Detection audit logs.",
+          "type": "string"
+        },
+        "storageAccountAccessKey": {
+          "description": "Specifies the identifier key of the Threat Detection audit storage account.",
+          "type": "string"
+        },
```
Comment (anuchandy) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: My understanding is ARM usually recommend to refer storage account using the fully qualified ARM resource id. This requires service to have linked access to storage that way service can reliably ensure there are sufficient permission to perform the operation (such as reading keys). Adding @ravbhatnagar from ARM team to comment.

**Review by anuchandy — COMMENTED** [review]
@ravbhatnagar please review the part referring storage account. Looks good from SDK side.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json (hunk 1 above)
Comment (ayeletshpigelman) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: It is similar in all our APIs for Postgres, SQL and Managed Instance
https://github.com/Azure/azure-rest-api-specs/pull/2947/files

**Comment by ayeletshpigelman** [comment]
@ravbhatnagar please review
It is urgent for Ignite
thank you!

**Comment by anuchandy** [comment]
@ayeletshpigelman whats your contact email? need to start an email thread with ARM with you in loop.

**Comment by ayeletshpigelman** [comment]
ayshpige@microsoft.com

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 2]
```diff
@@ -1062,6 +1062,126 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/securityAlertPolicies/{securityAlertPolicyName}": {
```
Comment (ravbhatnagar) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: Should there be an API to list the collection of securityPOlicies under a server.

**Review by ravbhatnagar — APPROVED** [review]
One minor note.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json (hunk 2 above)
Comment (ayeletshpigelman) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: There is only one policy, but the user can define its name.

**Comment by ayeletshpigelman** [comment]
@anuchandy can you please merge the pull request

**Review by anuchandy — APPROVED** [review]

**Pull request merged**